### PR TITLE
library.php improvements

### PR DESCRIPTION
**Pull request by jdub233**
This is a first pass at basic code quality improvements for the library.php collection of global functions, as  this file has a particularly low maintainability score in codeclimate.

There is a full phpcbf automatic formatting pass, and several basic WP standards improvements.  There is also a small amount of conditional refactoring, and some additional output escaping (thanks @stonybrookweb ). 

Mostly the changes are spacing and minor syntactic adjustments, no major code changes.

I've also added a round of functional documentation, the function descriptions and internal comments should help clarify how these functions operate and fit together.

The changes are centered particularly around the functions that the widget depends on, and includes some comments in the widget to inventory those functions.

### CODE REVIEW

**Comment by jdub233** [comment]
I ran visual regression tests against 3 BU sites (housing, law, bumc/devbio) successfully.  These functions also have phpunit test coverage, and these pass as well.

File: includes/library.php [hunk 1]
```diff
@@ -608,68 +720,75 @@ function bu_navigation_pages_by_parent_menu_sort_cb( $a, $b ) {
...
+		'anchor'       => true,
 		'title_before' => '',
-		'title_after' => '',
-		'section_ids' => null
-		);
-	$r = wp_parse_args( $args, $defaults );
+		'title_after'  => '',
+		'section_ids'  => null,
+	);
+	$r        = wp_parse_args( $args, $defaults );
```
Comment (rfera213) [inline includes/library.php:741]: Single letter variable could perhaps be refactored

Comment (jdub233) [reply includes/library.php:741]: Totally, I'm planning on additional cleanup releases

**Review by rfera213 — APPROVED** [review]
Looks good to me

**Pull request merged**